redis/asyncio/connection.py

```diff
@@ -685,7 +685,7 @@ def pack_commands(self, commands: Iterable[Iterable[EncodableT]]) -> List[bytes]
 
     def _socket_is_empty(self):
         """Check if the socket is empty"""
-        return not self._reader.at_eof()
+        return len(self._reader._buffer) == 0
```

Comment:
Let's link in here (as a comment) back to the PR. It's obvious enough from our discussion - but I'm worried that no one will use git blame?